### ISSUE TITLE
Crash With Deployment Build

### ISSUE THREAD

**Issue by AnotherCommander** [OoliteProject/oolite#132]
Would you believe it? Just after announcing our intention to release, here is the first crasher. It is actually quite a bad one.

Observed under Windows Vista 32-bit, using the 32-bit Deployment build of current trunk a0df280. The crash happens almost consistently. It is observable on a clean install of Oolite. To reproduce:
1. Load this save game: https://drive.google.com/file/d/0BwG6R5Qjd1f2NkIwd2FfeThYYlE/view?usp=sharing
2. Launch from station and travel a few seconds away from it.
3. Wait. The crash happens at a random point in time. Sometimes it may take a while. Insist if you don't get it at first. Try moving around a little. For me, it happens almost every time sooner or later, usually within a few minutes.

What is really bad about it, is that it seems to be trashing the stack. The only thing I get when I try to run it on gdb to get a backtrace is this (note the below is with the deployment executable, without stripping the debug info):
```


Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 3856.0x12f4]
0x02d1ef0f in ?? ()
(gdb) bt
#0  0x02d1ef0f in ?? ()
(gdb) Cannot access memory at address 0x3f5d0725
```
Hoping that I could get at least an event associated with the crash, I installed also the LogEvents OXP. The output associated with the above crash is below:
```
17:41:38.741 [LogEvents]: Player gui screen will change from GUI_SCREEN_LOAD to GUI_SCREEN_STATUS
17:41:38.742 [LogEvents]: Player gui screen changed from GUI_SCREEN_LOAD to GUI_SCREEN_STATUS
17:41:38.742 [LogEvents]: Player mission screen opportunity
17:42:11.171 [LogEvents]: Navigation Buoy 6698 spawned at 10 km
17:42:11.172 [LogEvents]: GalCop Viper Interceptor 8060 spawned at 32 km
17:42:11.172 [LogEvents]: Dodecahedron Station 30621 spawned at 0 km
17:42:11.172 [LogEvents]: Moray Star Boat 30775 spawned at 48 km
17:42:11.185 [LogEvents]: Asteroid 1127 spawned at 207 km
17:42:11.186 [LogEvents]: Asteroid 23401 spawned at 220 km
17:42:11.186 [LogEvents]: Rock Hermit 6965 spawned at 229 km
17:42:11.186 [LogEvents]: Asteroid 12783 spawned at 236 km
17:42:11.186 [LogEvents]: Asteroid 20367 spawned at 246 km
17:42:11.187 [LogEvents]: Asteroid 6132 spawned at 246 km
17:42:11.187 [LogEvents]: Asteroid 25632 spawned at 255 km
17:42:11.187 [LogEvents]: Asteroid 11806 spawned at 256 km
17:42:11.187 [LogEvents]: Asteroid 14069 spawned at 258 km
17:42:11.187 [LogEvents]: Rock Hermit 29354 spawned at 261 km
17:42:11.188 [LogEvents]: Asteroid 7078 spawned at 263 km
17:42:11.188 [LogEvents]: Asteroid 24612 spawned at 264 km
17:42:11.188 [LogEvents]: Asteroid 9375 spawned at 265 km
17:42:11.188 [LogEvents]: Asteroid 19399 spawned at 268 km
17:42:11.189 [LogEvents]: Asteroid 10153 spawned at 269 km
17:42:11.189 [LogEvents]: Asteroid 8154 spawned at 273 km
17:42:11.189 [LogEvents]: Asteroid 695 spawned at 275 km
17:42:11.189 [LogEvents]: Asteroid 10920 spawned at 276 km
17:42:11.189 [LogEvents]: Asteroid 10001 spawned at 339 km
17:42:11.190 [LogEvents]: Asteroid 16497 spawned at 357 km
17:42:11.190 [LogEvents]: Asteroid 17486 spawned at 357 km
17:42:18.897 [LogEvents]: Player gui screen changed from GUI_SCREEN_STATUS to GUI_SCREEN_OPTIONS
17:42:29.427 [LogEvents]: Player gui screen changed from GUI_SCREEN_GAMEOPTIONS to GUI_SCREEN_MAIN
17:42:29.429 [LogEvents]: Player will launch from Dodecahedron Station 30621 flying Cobra Mark III 9241
17:42:29.612 [LogEvents]: Player compass targeted lost target in mode COMPASS_MODE_PLANET
17:42:29.619 [LogEvents]: Player alert condition changed from 0 to 2
17:42:31.604 [LogEvents]: Player compass targeted Xevera undefined in mode COMPASS_MODE_PLANET
17:42:32.626 [LogEvents]: Player launched from Dodecahedron Station 30621
17:43:44.053 [LogEvents]: Adder 17283 spawned at 2 km
17:46:41.176 [LogEvents]: Cobra Mark III 7408 spawned at 76 km
17:46:41.176 [LogEvents]: Cobra Mark III 18692 spawned at 76 km
```

The crash happened a few seconds after the last log entry, so I don;t think that there is really any event logged that we can associate with it. It may be possible that the culprit is the save game itself - it is an old save that I use for general testing purposes and may contain something that the new Oolite doesn't like. I have not seen this happening with a new game yet but I will try more and see what happens. I will also try to reproduce on a Test Release build if I can and report back.

**Comment by AnotherCommander** [comment]
Tried the exact same executable and save game on my work computer sporting Win7 54-bit and an Intel HD card. Result: So far I am unable to reproduce the crash. I am trying to get some testing feedback from Cody in the forums too. If he cannot crash it either, then I guess it;'s something to do with my system at home. I will have to re-check my NVidia settings as well; I seem to recall Threaded Optimization causing crashes in the past and I am not sure what it is set to right now.

**Comment by kanthoney** [comment]
I've not been able to reproduce this on my Windows 7 box with nvidia card so far.  Mind you, I'm not quite sure how to make a deployment release (from memory, I tried make -f Makefile release-deployment with a 32 bit build environment) so I might have done that wrong.

**Comment by AnotherCommander** [comment]
No, that's exactly how you can generate the deployment build that is distributed in release. I really hope it's something silly like a gfx setting on my side so we can close this asap.

**Comment by cim--** [comment]
Works fine on the Linux deployment build, though this sort of crash I wouldn't expect to be cross-platform.

**Comment by kanthoney** [comment]
Can you make the executable available?  It might be worth trying with that.

**Comment by AnotherCommander** [comment]
Guys, it's got to be the Threaded Optimization option from the Nvidia Control Panel. I disabled it, started the game, went to dinner, came back after an hour, still running. Enabled TO, run the game, crash within one minute. Disabled TO, run game, it's been running now for at least ten minutes and still running as I am writing this.

If anone wants to test, here is the link to the exe+save: https://drive.google.com/file/d/0BwG6R5Qjd1f2T1hPNjFrdTZoRUE/view?usp=sharing

Kevin, do you have Threading Optimization enabled on your NVidia? If not, it may be interesting to enable and try running the exe again.

I'll test some more, but this one looks like it's heading for quick closure. Cody reported long test running already (TO is off for him too).

**Comment by kanthoney** [comment]
TO was set to "auto", so I'm not sure whether it was on or off for Oolite.  It's set to "on" now - no crashes as yet after a few minutes but I'll leave it running.  NVidia driver version is 347.88.

There were also no crashes on an extremely old XP machine with Intel driver that I dug out.

**Comment by kanthoney** [comment]
No crash with TO set to on, either with the executable built by me or with the one provided.  Trying TO off instead.

**Comment by kanthoney** [comment]
There's been a complete lack of crashing here, with TO on or off, or with a 32-bit or 64-bit deployment build.

**Comment by AnotherCommander** [comment]
Good. Probably it is dependent on driver version then. So, let's summarize:
We have tested this on 5 computers so far: 
1 . My Vista 32-bit NVidia 9600GTM laptop.
2. My Win 7 64-bit Intel HD laptop.
3. Kevin's Win 7 w/ NVidia box.
4. Kevin's old XP box.
5. Cody's Win 7 64-bit NVidia box.

To this, we could add Chris' Linux box, but let's keep it oriented to Windows for now, since it works on Linux anyway.

Four of the five cases had no problems running the deployment build. Of the four, one (3) was tested with and without TO, no difference to result. No issues encountered.

In one case (1) we encountered persistent crash on deployment build, This problem was fixed upon disanling Threaded Optimization on the driver.

My conclusion is that certain driver versions / NVidia card combinations may result in a buggy implementation of Threaded Optimization which can be expressed as crashes. The solution is to disable this option. By doing that, all our cases become clean and work as intended.

Resolution: Not a bug, will not fix. Let's just keep it in mind in case of possible future reports about crashes on Windows. Issue closed.
